libraries/classes/Table.php

```diff
@@ -542,7 +543,8 @@ public static function generateFieldSpec(
         // see https://dev.mysql.com/doc/refman/5.5/en/floating-point-types.html
         $pattern = '@^(DATE|TINYBLOB|TINYTEXT|BLOB|TEXT|'
             . 'MEDIUMBLOB|MEDIUMTEXT|LONGBLOB|LONGTEXT|SERIAL|BOOLEAN|UUID|JSON)$@i';
-        if (strlen($length) !== 0 && ! preg_match($pattern, $type)) {
+        if (strlen($length) !== 0 && ! preg_match($pattern, $type)
+        && Compatibility::isIntegerTypesSupportLength($type, strlen($length), $dbi)) {
```

Comment:
```suggestion
        if (
            strlen($length) !== 0
            && ! preg_match($pattern, $type)
            && Compatibility::isIntegerTypesSupportLength($type, strlen($length), $dbi)
        ) {
```

Maybe check `composer run phpcbf/phpcs` after that ;)

Reply:
Just a micro optimization, `strlen($length)` could be extracted to a variable to be used only once

Reply:
`phpcbf` didn't return any errors, `phpcs` exhausted the memory, I will try to re-run it when I figured out how to increase the allowed memory

Reply:
`php -d memory_limit=-1 ./vendor/bin/phpcs` could work

Reply:
Thanks, no errors returned  🎉